packages/devtools_app/lib/src/service_manager.dart

```diff
@@ -809,7 +813,7 @@ class ServiceExtensionManager extends Disposer {
   }
 
   Future<void> _maybeAddServiceExtension(String name) async {
-    if (_firstFrameEventReceived || !isUnsafeBeforeFirstFlutterFrame(name)) {
+    if (_firstFrameEventReceived /*|| !isUnsafeBeforeFirstFlutterFrame(name)*/) {
```

Comment:
why remove the check to `isUnsafeBeforeFirstFlutterFrame`?

Reply:
This change is more for illustrating the problem than fixing it - it makes devtools work if paused on start and then hot-restarted. 

If the user connects devtools to a paused application and hot-restarts the app, `_maybeAddServiceExtension` does not wait for the isolate to finish restarting and proceeds to re-register the extensions with an old isolate id.

I am just forcing the `else` clause to execute here instead of `then` and add registrations as pending after isolate exit.

Does it seem like a bug in devtools, or is this something that dwds doing incorrectly?

Reply:
seems like a bug in DevTools.